File: LeetCode/2309. Greatest English Letter in Upper and Lower Case/solution.py
"""
Runtime: 28 ms
Memory Usage: 15.1 MB
"""
class Solution:
    def greatestLetter(self, s: str) -> str:
        rec = set(s)
        for l, u in zip(reversed(ascii_lowercase), reversed(ascii_uppercase)):
            if l in rec and u in rec:
                return u
        return ""

File: LeetCode/2309. Greatest English Letter in Upper and Lower Case/solution.cpp
/**
 * Runtime: 0 ms
 * Memory Usage: 7.7 MB 
 */
class Solution {
public:
    string greatestLetter(string s) {
        unordered_set<char> rec(s.begin(), s.end());
        for (int i = 25; i >= 0; -- i) {
            if (rec.count('a' + i) && rec.count('A' + i)) {
                return string(1, 'A' + i);
            }
        }
        return "";
    }
};
